tests/beacon/test_helpers.py

```diff
@@ -707,6 +722,224 @@ def test_get_domain(pre_fork_version,
+    votes = SlashableVoteData(**params)
+    return votes.messages
+
+
+def test_verify_slashable_vote_data_signature(privkeys,
+                                              sample_beacon_state_params,
+                                              genesis_validators,
+                                              sample_slashable_vote_data_params):
+    sample_beacon_state_params["validator_registry"] = genesis_validators
+    state = BeaconState(**sample_beacon_state_params)
+
+    # NOTE: we can do this before "correcting" the params as they
+    # touch disjoint subsets of the provided params
+    messages = _create_slashable_vote_data_messages(sample_slashable_vote_data_params)
+
+    valid_params = _correct_slashable_vote_data_params(sample_slashable_vote_data_params,
+                                                       genesis_validators,
+                                                       messages,
+                                                       privkeys)
+    valid_votes = SlashableVoteData(**valid_params)
+    assert verify_slashable_vote_data_signature(state, valid_votes)
+
+    invalid_params = _corrupt_signature(valid_params)
+    invalid_votes = SlashableVoteData(**invalid_params)
+    assert not verify_slashable_vote_data_signature(state, invalid_votes)
+
+
+def _run_verify_slashable_vote(params, state, max_casper_votes, should_fail):
+    votes = SlashableVoteData(**params)
+    result = verify_slashable_vote_data(state, votes, max_casper_votes)
+    if should_fail:
+        assert not result
+    else:
+        assert result
+
+
+@pytest.mark.parametrize(
+    (
+        'param_mapper,'
+        'should_fail'
+    ),
+    [
+        (lambda params: params, False),
+        (lambda params: _corrupt_vote_count(params), True),
+        (lambda params: _corrupt_signature(params), True),
+        (lambda params: _corrupt_vote_count(_corrupt_signature(params)), True),
+    ],
+)
+def test_verify_slashable_vote_data(param_mapper,
+                                    should_fail,
+                                    privkeys,
+                                    sample_beacon_state_params,
+                                    genesis_validators,
+                                    sample_slashable_vote_data_params,
+                                    max_casper_votes):
+    sample_beacon_state_params["validator_registry"] = genesis_validators
+    state = BeaconState(**sample_beacon_state_params)
+
+    # NOTE: we can do this before "correcting" the params as they
+    # touch disjoint subsets of the provided params
+    messages = _create_slashable_vote_data_messages(sample_slashable_vote_data_params)
+
+    params = _correct_slashable_vote_data_params(sample_slashable_vote_data_params,
+                                                 genesis_validators,
+                                                 messages,
+                                                 privkeys)
```

Comment:
nitpick of formatting: https://github.com/ethereum/snake-charmers-tactical-manual/blob/master/style-guide.md#closing-parenthesisbracketbrace